### PR TITLE
[docs] Document middleware response changes

### PR DESCRIPTION
**Pull request by ethomson**
Beginning in Next.js 13, middleware can now respond directly, without needing to rewrite or redirect, when the experimental `.allowMiddlewareResponseBody` configuration setting is set. Document this behavior.

## Bug

- [ ] Related issues linked using `fixes #number`
- [ ] Integration tests added
- [ ] Errors have a helpful link attached, see `contributing.md`

## Feature

- [ ] Implements an existing feature request or RFC. Make sure the feature request has been accepted for implementation before opening a PR.
- [ ] Related issues linked using `fixes #number`
- [ ] Integration tests added
- [ ] Documentation added
- [ ] Telemetry added. In case of a feature if it's used or not.
- [ ] Errors have a helpful link attached, see `contributing.md`

## Documentation / Examples

- [x] Make sure the linting passes by running `pnpm build && pnpm lint`
- [ ] The "examples guidelines" are followed from [our contributing doc](https://github.com/vercel/next.js/blob/canary/contributing/examples/adding-examples.md)

### CODE REVIEW

File: docs/advanced-features/middleware.md [hunk 1]
```diff
@@ -218,6 +218,44 @@ export function middleware(request: NextRequest) {
...
+
+```js:next.config.js
+module.exports = {
+  experimental: {
+    allowMiddlewareResponseBody: true
+  },
+};
+```
+
+Once enabled, you can provide a response from middleware using the `NextResponse` API:
```
Comment (ijjk) [inline docs/advanced-features/middleware.md:235]: This can also be a standard response e.g. a `fetch` `Response`.

Comment (ethomson) [inline docs/advanced-features/middleware.md:235]:
```suggestion
Once enabled, you can provide a response from middleware using the `Response` or `NextResponse` API:
```

Comment (ethomson) [reply docs/advanced-features/middleware.md]: updated

File: docs/advanced-features/middleware.md [hunk 2]
```diff
@@ -9,14 +9,14 @@ description: Learn how to use Middleware to run code before a request is complet
...
-| `v13.0.0` | Support overriding request headers.                                                        |
+| `v13.0.0` | Middleware can modify request headers, response headers, and send responses                |
 | `v12.2.0` | Middleware is stable                                                                       |
 | `v12.0.9` | Enforce absolute URLs in Edge Runtime ([PR](https://github.com/vercel/next.js/pull/33410)) |
 | `v12.0.0` | Middleware (Beta) added                                                                    |
 
 </details>
 
-Middleware allows you to run code before a request is completed, then based on the incoming request, you can modify the response by rewriting, redirecting, adding headers, or setting cookies.
+Middleware allows you to run code before a request is completed, then based on the incoming request, you can modify the response by rewriting, redirecting, modifying the request or response, or responding directly.
```
Comment (S3Prototype) [inline docs/advanced-features/middleware.md:19]: Is it not worth using the word "headers" here to be explicit? Genuine question. If it's not necessary and would just make the wording clunky, that's cool

Comment (ethomson) [inline docs/advanced-features/middleware.md:19]:
```suggestion
Middleware allows you to run code before a request is completed, then based on the incoming request, you can modify the response by rewriting, redirecting, modifying the request or response headers, or responding directly.
```

Comment (ethomson) [reply docs/advanced-features/middleware.md]: good suggestion, thanks!

File: docs/advanced-features/middleware.md [hunk 3]
```diff
@@ -218,6 +218,44 @@ export function middleware(request: NextRequest) {
 
 > **Note:** Avoid setting large headers as it might cause [431 Request Header Fields Too Large](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/431) error depending on your backend web server configuration.
 
+## Producing a Response
+
+You can respond to middleware directly by returning a `NextResponse` (responding from middleware is available since Next.js v13.0.0).
+
+To enable middleware responses, update `next.config.js`:
+
+```js:next.config.js
```
Comment (ijjk) [inline docs/advanced-features/middleware.md:227]: This isn't valid here and should be on a new line e.g.
```suggestion
```js
// next.config.js
```

**Review by ijjk — APPROVED** [review]

**Pull request merged**